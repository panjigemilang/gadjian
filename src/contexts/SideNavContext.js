import { createContext, useState } from "react"

export const SideNavContext = createContext()

const SideNavContextProvider = (props) => {
  const [toggleSideNav, setToggleSideNav] = useState(false)

  return (
    <SideNavContext.Provider value={{ toggleSideNav, setToggleSideNav }}>
      {props.children}
    </SideNavContext.Provider>
  )
}

export default SideNavContextProvider
