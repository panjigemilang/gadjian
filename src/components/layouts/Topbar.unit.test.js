import { mount } from "enzyme"
import Topbar from "./TopBar"
import { SideNavContext } from "../../contexts/SideNavContext"
import toJson from "enzyme-to-json"

describe("<Topbar />", () => {
  it("should render Topbar properly", () => {
    const defaultValues = {
      toggleSideNav: false,
      setToggleSideNav: jest.fn(),
    }

    const wrapper = mount(
      <SideNavContext.Provider value={defaultValues}>
        <Topbar />
      </SideNavContext.Provider>
    )

    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })
})
