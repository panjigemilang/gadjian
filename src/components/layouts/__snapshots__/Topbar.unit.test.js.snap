// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Topbar /> should render Topbar properly 1`] = `
<TopBar>
  <header
    className="topbar-wrapper"
  >
    <div
      className="user-wrapper"
    >
      <div
        className="burger-menu"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
      <div
        className="mobile-logo"
      >
        <img
          alt="Logo"
          src="logo.png"
        />
      </div>
      <h2
        className="small-hide"
      >
        Hallo, 
        <span>
          Gadjian User
        </span>
      </h2>
      <img
        alt="User"
        className="profile-thumbnail"
        src="http://via.placeholder.com/80x80.png"
      />
    </div>
  </header>
</TopBar>
`;
